shopify/utils/exceptions.py

```diff
@@ -0,0 +1,10 @@
+class InvalidIssuerError(Exception):
+    pass
+
+
+class MismatchedHostsError(Exception):
+    pass
+
+
+class TokenAuthenticationError(Exception):
+    pass
```

Comment:
I wonder if we should either just use one exception class, or have a parent class that these inherit from. The value of that is that it enables adopters to catch all session token validation related errors with one catch block.

To fully realize the benefits of this, we'd probably also want to catch whatever errors PyJWT raises and re-raise so that they're under the same exception hierarchy. There's a handy `raise <new exception> from <some other exception>` format that ensures the original stack trace is preserved.

Reply:
I like having the shared base class for exceptions as it helps callers know whether the exception is coming from us or not.